drivers/i2c/i2c_nrfx_twim.c

```diff
@@ -297,40 +297,26 @@ static const struct i2c_driver_api i2c_nrfx_twim_driver_api = {
 	.recover_bus = i2c_nrfx_twim_recover_bus,
 };
 
-#ifdef CONFIG_PM_DEVICE
 static int twim_nrfx_pm_action(const struct device *dev,
 			       enum pm_device_action action)
 {
 	const struct i2c_nrfx_twim_config *dev_config = dev->config;
-	int ret = 0;
 
 	switch (action) {
 	case PM_DEVICE_ACTION_RESUME:
-		ret = pinctrl_apply_state(dev_config->pcfg,
-					  PINCTRL_STATE_DEFAULT);
-		if (ret < 0) {
-			return ret;
-		}
+		(void)pinctrl_apply_state(dev_config->pcfg, PINCTRL_STATE_DEFAULT);
```

Comment:
why to resign from returned error codes?

Reply:
Because it is not fallible in the case of Nordic drivers. Well, it theoretically can if users set eg pinctrl-names = "defaultx". I can leave it if preferred.